gensim/models/wrappers/ldamallet.py

```diff
@@ -245,11 +248,40 @@ def show_topic(self, topicid, topn=10):
     def print_topic(self, topicid, topn=10):
         return ' + '.join(['%.3f*%s' % v for v in self.show_topic(topicid, topn)])
 
+
+    def get_version(self, direc_path):
+        """"
+
+        function to return the version of `mallet`
+
+        """
+        try:
+            """
+            Check version of mallet via jar file
+            """
+            archive = zipfile.ZipFile(direc_path, 'r')
```

Comment:
Yes, according to the following eg in the smart_open repo's read me:
for line in smart_open.smart_open('./foo.txt.gz'):
...    print line

We can only  unzip a file (say a txt file).
I think zipfile is OK. If there not then I will look for another library to handle jar file.